### PR TITLE
Issue #59: Display the connector variables

### PR DESCRIPTION
**Pull request by CherfaElyes**
* Added a `variables` section under the `prerequisites` in the connectors pages to display the connector template variables, their description & their default value if available.
* Updated MetricsHub example content producer to add variables configuration example.
* Tested on metricshub-doc.

# Tests
## Default connector variables specified (description and defaultValue specified)
image

## Default connector variables specified (description not specified)
image

## Default connector variables specified (defaultValue not specified)
image

## Default connector variables not specified (variables are present).
image

## MetricsHub.yaml example
image

### CODE REVIEW

**Comment by bertysentry** [comment]
I would simplify the listing of variables like below:

Variables:

* `matchCommand`: Regular expression that command line of the process must match with (default: `.*`)
* `matchName`: Regular expression that the process name must match with (default: `.*`)

So the "template" is like:

```html
<li> <code>VARNAME</code>: VARDESCRIPTION (default: <code>DEFAULTVALUE</code>)
```

Description and default value should be displayed in the doc only if they are specified, like below:

* `matchName`
* `matchName`: Regular expression that the process name must match with
* `matchName` (default: `.*`)

**Comment by bertysentry** [comment]
Now an interesting question: how do we specify variable values for a connector through the CLI? 😅

**Comment by NassimBtk** [comment]
There is a plan for this @bertysentry https://github.com/sentrysoftware/metricshub/issues/357

File: src/main/java/org/sentrysoftware/maven/metricshub/connector/producer/ConnectorJsonNodeReader.java [hunk 1]
```diff
@@ -605,4 +608,56 @@ public List<String> getTags() {
...
+	 * Variables are expected to be in the format: ${var::variableName}.
+	 *
+	 * @return a set of unique variable names found within the connector template.
+	 */
+	public Set<String> getVariablesNames() {
+		final String stringConnector = connector.toString();
+		final String variableRegex = "\\$\\{var::(.*?)\\}";
+		final Set<String> variables = new HashSet<>();
+
+		final Pattern pattern = Pattern.compile(variableRegex);
```
Comment (NassimBtk) [inline src/main/java/org/sentrysoftware/maven/metricshub/connector/producer/ConnectorJsonNodeReader.java:623]: Create a constant please.

File: src/main/java/org/sentrysoftware/maven/metricshub/connector/producer/model/common/ConnectorDefaultVariable.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package org.sentrysoftware.maven.metricshub.connector.producer.model.common;
+
+import lombok.AllArgsConstructor;
```
Comment (NassimBtk) [inline src/main/java/org/sentrysoftware/maven/metricshub/connector/producer/model/common/ConnectorDefaultVariable.java:3]: Should be under the ` ╱╲╱╲╱╲`  header comment.

**Review by NassimBtk — CHANGES_REQUESTED** [review]

**Review by bertysentry — APPROVED** [review]

**Review by NassimBtk — APPROVED** [review]

**Pull request merged**